### PR TITLE
Feature: added import transformer for massaExport functions

### PR DESCRIPTION
**Pull request by 0xSwapFeeder**
This transformer allows to use massa exported function in other files in the same contract.

For the moment only the build is tested, I still need to test on sample contracts that the feature works.

I would like other people to test it as well to avoid the 'it works on my amchine' syndrome.

### CODE REVIEW

**Comment by 0xSwapFeeder** [comment]
Broken, on hold.

**Comment by 0xSwapFeeder** [comment]
In testing, but can start reviewing and testing on your side to avoid "it works on my machine" syndrome.

To do so, checkout this branch, create a smart contract, and in asconfig.json add a transformer locally : e.g.: 
```json
        "release": {
            "outFile": "build/main.wasm",
            "sourceMap": true,
            "optimizeLevel": 3,
            "shrinkLevel": 0,
            "converge": false,
            "noAssert": false,
            "transform": [
                "../../../as/packages/as-transformer"
            ]
```

File: packages/as-transformer/src/helpers/source.ts [hunk 1]
```diff
@@ -12,10 +12,11 @@ export function parseFile(
     const shortFilePath = filePath.replace(/.*node_modules/i, '~lib');
     parser.parseFile(readFileSync(filePath, 'utf-8'), shortFilePath, false);
   } else {
+    subDir;
```
Comment (gregLibert) [inline packages/as-transformer/src/helpers/source.ts:15]: What this is doing?

File: packages/as-transformer/src/transformer.ts [hunk 2]
```diff
@@ -85,7 +64,13 @@ export class Transformer extends TransformVisitor {
     const inputText = (node.expression as IdentifierExpression)?.text;
 
     for (let transformer of callTransformers) {
+      if (
+        (transformer as MassaExportCalls) !== undefined &&
+        this.originalVisiting
+      )
```
Comment (gregLibert) [inline packages/as-transformer/src/transformer.ts:70]: Can you add a comment explaining why you continue here?

File: packages/as-transformer/src/transformer.ts [hunk 3]
```diff
@@ -206,33 +164,43 @@ export class Transformer extends TransformVisitor {
...
+              (update.from === 'MassaExport' &&
+                (source.internalPath.includes(
+                  update.data.get('funcToPrivate')![0] + 'Helper',
+                ) ||
+                  source.internalPath.includes(
+                    update.data.get('funcToPrivate')![0] + 'RHelper',
+                  ))) ||
+              (update.from === 'as-transformer' &&
+                source.internalPath.includes(update.content)),
+          ).length <= 0 && */
```
Comment (gregLibert) [inline packages/as-transformer/src/transformer.ts:185]: Why is this commented? If you want to keep it, please add a comment explaining why.

File: packages/as-transformer/src/transformer.ts [hunk 4]
```diff
@@ -247,26 +215,22 @@ export class Transformer extends TransformVisitor {
...
         // Updating original file source
         actualSource = this._updateSource(actualSource, newContent, parser);
-        updatedSources.push(source);
         transformer.resetUpdates();
-        this.visit(actualSource);
       }
-      const update: Update = {
+      updatedSources.push(actualSource);
+      updatedSources.forEach((s) => this.visit(s));
+      /* const update: Update = {
```
Comment (gregLibert) [inline packages/as-transformer/src/transformer.ts:222]: Alsor true here.

File: packages/as-transformer/src/transformer.ts [hunk 5]
```diff
@@ -206,33 +164,43 @@ export class Transformer extends TransformVisitor {
    * @param parser - A {@link Parser} object.
    */
   afterParse(parser: Parser): void {
+    Debug.log('Starting after parse.');
```
Comment (gregLibert) [inline packages/as-transformer/src/transformer.ts:167]:
```suggestion
    Debug.log('Starting after parse hook.');
```

File: packages/as-transformer/src/transformer.ts [hunk 6]
```diff
@@ -206,33 +164,43 @@ export class Transformer extends TransformVisitor {
...
+          ).length <= 0 && */
+          !source.internalPath.includes('build/')
+        );
+      },
     );
     let updatedSources: Source[] = [];
 
-    if (sources.length <= 0) return;
+    if (sources.length <= 0) {
+      Debug.log('No sources to transform, exit');
```
Comment (gregLibert) [inline packages/as-transformer/src/transformer.ts:193]:
```suggestion
      Debug.log('No sources to transform, exit after parse hook');
```

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: packages/as-transformer/src/transformer.ts (hunk 2 above)
Comment (gregLibert) [reply packages/as-transformer/src/transformer.ts]: I don't see any explanaition. Did I miss something?

File: packages/as-transformer/src/transformer.ts (hunk 3 above)
Comment (gregLibert) [reply packages/as-transformer/src/transformer.ts]: same here

File: packages/as-transformer/src/transformer.ts (hunk 4 above)
Comment (gregLibert) [reply packages/as-transformer/src/transformer.ts]: Same here.

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: packages/as-transformer/src/transformer.ts (hunk 2 above)
Comment (0xSwapFeeder) [reply packages/as-transformer/src/transformer.ts]: That's odd, on my side I pushed changes.

Comment (0xSwapFeeder) [reply packages/as-transformer/src/transformer.ts]: You don't see them @gregLibert ?

**Review by gregLibert — APPROVED** [review]

**Pull request merged**